src/MediaHandler.js

```diff
@@ -324,12 +335,24 @@ export default class MediaHandler {
       },
       cache: 'no-store',
       signal: timeoutSignal(this._fetchTimeout),
+      retryOn: (attempt, error, response) => {
+        // if the fetch timeout exceeded, stop now
+        if (error instanceof AbortError) {
+          return false;
+        }
+        // retry on any other network error or 5xx status codes
+        if (attempt < MAX_RETRIES && (error !== null || response.status >= 500)) {
+          log.debug(`failed with ${error || response.status}: retrying (attempt# ${attempt + 1}/${MAX_RETRIES})`);
```

Comment:
maybe log the fetched url?

Reply:
it already logs the fetched URL on line 327 above, also with level `debug`. 

Maybe, log the URL and failure level `info` instead, or is that too chatty?

Reply:
yes, too chatty